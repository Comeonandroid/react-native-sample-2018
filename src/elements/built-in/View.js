import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'


const View = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

View.propTypes = {
  children: PropTypes.node
}

const Element = styled.View `
`

export default View
