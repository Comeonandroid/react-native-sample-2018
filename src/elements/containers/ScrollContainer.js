import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'
import { ifProp, prop } from 'styled-tools'

import View from '../built-in/View'
import ScrollView from '../built-in/ScrollView'

class ScrollContainer extends PureComponent {

  constructor(props) {
    super(props)

    this.setHeight = this.setHeight.bind(this)
  }

  setHeight() {
    const { fullheight } = this.props
    return fullheight ? { flexGrow: 1 } : {}
  }

  render() {
    const { children, disabled, ...rest } = this.props
    return (
      <Element contentContainerStyle={this.setHeight()} {...rest}>
        {children}
        {disabled && <Overlay show={disabled} />}
      </Element>
    )
  }
}

ScrollContainer.propTypes = {
  children: PropTypes.node,
  fullheight: PropTypes.bool,
  disabled: PropTypes.bool
}

ScrollContainer.defaultProps = {
  disabled: false,
  fullheight: false
}

const Element = styled(ScrollView) `
  position: relative;
`

const Overlay = styled(View) `
  height: ${prop('height', 'auto')};
  display: ${ifProp('show', 'flex', 'none')};
  position: absolute;
  flex: 1;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 9999;
  background: #fff;
  opacity: 0.6;
`

export default ScrollContainer