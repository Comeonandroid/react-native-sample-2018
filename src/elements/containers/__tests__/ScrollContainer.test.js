import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import View from '../../built-in/View'
import ScrollContainer from '../ScrollContainer'

describe('<ScrollContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <ScrollContainer /> })

  describe('fullheight: true', () => {
    itBehavesLike('aThemedComponent', { subject: <ScrollContainer fullheight /> })
  })

  describe('disabled: false', () => {
    it('onPress works for passed children', () => {
      const onPress = jest.fn()
      const wrapper = shallow(<ScrollContainer>
        <View onPress={onPress} />
      </ScrollContainer>)
      wrapper.find(View).simulate('press')
      expect(onPress).toHaveBeenCalledTimes(1)
    })
  })

  describe('disabled: true', () => {
    itBehavesLike('aThemedComponent', { subject: <ScrollContainer disabled /> })
    it('onPress blocked for passed children', () => {
      const onPress = jest.fn()
      const wrapper = shallow(<ScrollContainer>
        <View onPress={onPress} />
      </ScrollContainer>)
      wrapper.find(View).simulate('press')
      expect(onPress).toHaveBeenCalledTimes(1)
    })
  })
})
