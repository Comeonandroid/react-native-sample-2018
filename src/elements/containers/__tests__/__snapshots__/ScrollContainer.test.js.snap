// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollContainer /> dark: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  fullheight={false}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`<ScrollContainer /> disabled: true dark: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  fullheight={false}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View>
    <View
      show={true}
      style={
        Array [
          Object {},
          Array [
            Object {
              "backgroundColor": "#fff",
              "bottom": 0,
              "display": "flex",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "auto",
              "left": 0,
              "opacity": 0.6,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 9999,
            },
            undefined,
          ],
        ]
      }
    />
  </View>
</RCTScrollView>
`;

exports[`<ScrollContainer /> disabled: true light: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  fullheight={false}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View>
    <View
      show={true}
      style={
        Array [
          Object {},
          Array [
            Object {
              "backgroundColor": "#fff",
              "bottom": 0,
              "display": "flex",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "auto",
              "left": 0,
              "opacity": 0.6,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 9999,
            },
            undefined,
          ],
        ]
      }
    />
  </View>
</RCTScrollView>
`;

exports[`<ScrollContainer /> fullheight: true dark: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  fullheight={true}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`<ScrollContainer /> fullheight: true light: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  fullheight={true}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View />
</RCTScrollView>
`;

exports[`<ScrollContainer /> light: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  fullheight={false}
  style={
    Array [
      Object {},
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ],
    ]
  }
>
  <View />
</RCTScrollView>
`;
